persistence/src/commonTest/kotlin/com/wire/kalium/persistence/backup/RestoreBackupTest.kt

```diff
@@ -0,0 +1,487 @@
+
+    @BeforeTest
+    fun setUp() {
+        deleteDatabase()
+        userDatabaseBuilder = createDatabase()
+        userDatabaseDataGenerator = UserDatabaseDataGenerator(
+            userDatabaseBuilder = userDatabaseBuilder,
+            databasePrefix = "user"
+        )
+
+        // TODO: using UserIDEntity is not correct, because the file will contain another user id
+        // which we do want to preserve when restoring from the user database
+        deleteDatabase(backupUserIdEntity)
+        backupDatabaseBuilder = createDatabase(backupUserIdEntity)
+        backupDatabaseDataGenerator = UserDatabaseDataGenerator(
+            userDatabaseBuilder = backupDatabaseBuilder,
+            databasePrefix = "backup"
+        )
+    }
+
+    @Test
+    fun givenBackupHasUniqueConversationAndUserHasUniqueConversations_whenRestoringBackup_TheBackupConversationAreInsertedInUserDatabase() =
+        runTest {
+            // given
+            val backupConversationAmount = 100
+
+            val conversationsToBackup = backupDatabaseDataGenerator.generateAndInsertConversations(
+                conversationAmount = backupConversationAmount,
+                messagePerConversation = 10,
+                messageType = MessageType.Regular
+            )
+
+            val userConversationAmount = 50
+            userDatabaseDataGenerator.generateAndInsertConversations(
+                conversationAmount = userConversationAmount,
+                messagePerConversation = 5,
+                messageType = MessageType.Regular
+            )
+            // when
+            userDatabaseBuilder.backupImporter.importFromFile(databasePath(backupUserIdEntity))
+
+            // then
+            val conversationsAfterBackup: List<ConversationEntity> = userDatabaseBuilder.conversationDAO.getAllConversations().first()
+
+            assertTrue(conversationsAfterBackup.containsAll(conversationsToBackup))
+            assertEquals(backupConversationAmount + userConversationAmount, conversationsAfterBackup.size)
+        }
+
+    @Test
+    fun givenBackupHasUnqiueConversationsAndSomeOverLappingWithUser_whenRestoringBackup_thenOnlyUnqiueConversationAreRestored() = runTest {
+        // given
+        val uniqueUserConversationAmount = 25
+        val userConversations = userDatabaseDataGenerator.generateAndInsertConversations(
+            conversationAmount = uniqueUserConversationAmount,
+            messagePerConversation = 5,
+            messageType = MessageType.Regular
+        )
+
+        val uniqueBackupConversationAmount = 25
+        val uniqueBackupConversations = backupDatabaseDataGenerator.generateAndInsertConversations(
+            conversationAmount = uniqueBackupConversationAmount,
+            messagePerConversation = 10,
+            messageType = MessageType.Regular
+        )
+
+        backupDatabaseBuilder.conversationDAO.insertConversations(userConversations)
```

Comment:
I am not sure if I follow, 

I wanted to test here that if the conversations are not overlapping the total amount of the conversation after restoring them will not be 75 but will be equal to the unique backup conversation and the user conversations

Reply:
Where is this 75 coming from?

Reply:
This line:
`backupDatabaseBuilder.conversationDAO.insertConversations(userConversations)` is inserting the `userConversations` on the backup database to be imported. But these conversations have also been previously inserted on the `userDatabase` with the method `userDatabaseDataGenerator.generateAndInsertConversations(...)`. So they are overlapping each other DB. I don't think this is what we want to test.

Reply:
exactly so they are overlapping which means if we restore the backup we do not want them to be inserted into the user data base, therefore if they are not inserted the total size must be equal to unique users one + unique backup ones

Reply:
if you know any better way to validate it, let me know, but that is what I came up with

Reply:
we also check if the conversations contain the unique backup ones so the remaining ones must be the one that were the overlapping and we are sure that they are not inserted twice

Reply:
Does it make any difference ?

Reply:
Well, I think it does, as it covers one more possible case. Also it would be good for coherence, as that way it will make what the test name states it does indeed :)

Reply:
Well, I think it does, as it covers one more possible case. Also it would be good for coherence, as that way it will make what the test name states it does indeed :)

Reply:
Well, I think it does, as it covers one more possible case. Also it would be good for coherence, as that way it will make what the test name states it does indeed :)

Reply:
Why does it makes a difference ? Either we insert all the converations that the user has into the backup db, or just 2, the test remains the same, check if those that are in both db. files are duplicated or not

Reply:
maybe I do not get something, let me know

Reply:
I got the point, thanks for the explanation I will add another unit tests